### PR TITLE
Merge dlme-traject into dlme-transform

### LINKED ISSUE #221: Improve integration testing to include dlme-traject
Add a task to circle ci to exercise the docker dlme-transform docker task with select (or all) of the dlme-traject based configurations in order to avoid update based breaking changes.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes sul-dlss/dlme-traject#146
Fixes #221 

Includes:
* Stop ignoring `config/` directory, since it will hold configuration like all
our other codebases do.
* Merge in Rubocop configuration from `dlme-traject`
* Remove all vestiges of `SKIP_FETCH_CONFIG` which is no longer a relevant
environment variable because configs now live under version control
* Update README documentation
* Move `settings.yml` and `metadata_mapping.json` into the `config/` directory
* Copy Traject configs into `traject_configs/`, to keep them from being confused
with gem configuration
* Update specs

Also includes:
* Add integration test that makes sure transform macros and rules work with Traject configs
* Set transformer configs in `config/settings.yml` so they are accessible across the codebase
* Use `config/settings.yml` settings in `Dlme::CLI`
* Allow `ConfigFinder` to return only a sampling of configs: this is what the integration test uses to test a subset of the DLME dataset
* Return dataset files sorted alphabetically (this lets us avoid some extremely large CSV data)
* Set `close_output_on_close` Traject option in `Dlme::Transformer` (for sul-dlss/dlme-traject#146)
* Comment out `cho_date_range_norm` field mappings in Traject configs to get tests to pass
* Pull latest `fgdc_config` from soon-to-be-defunct dlme-traject codebase
* Remove `lang('ar-Latn')` from Harvard Traject config (I added this for testing and mistakenly committed it)
* Allow resetting the `RecordCounter` (for the test suite)

## Why was this change made?

To streamline development practices and facilitate more robust testing of Traject configs (#221).

## Was the documentation (README, API, wiki, ...) updated?

Yes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,81 +5,87 @@
 [![Docker Container](https://images.microbadger.com/badges/image/suldlss/dlme-transform.svg)](https://microbadger.com/images/suldlss/dlme-transform "docker container image")
 [![Latest Commit](https://images.microbadger.com/badges/commit/suldlss/dlme-transform.svg)](http://microbadger.com/images/suldlss/dlme-transform "latest commit")
 
-Transforms raw DLME metadata from https://github.com/sul-dlss/dlme-metadata and uses transformations in https://github.com/sul-dlss/dlme-traject to create [DLME intermediate representation](https://github.com/sul-dlss/dlme/blob/master/docs/application_profile.md) documents in S3.
+Transforms raw DLME metadata from https://github.com/sul-dlss/dlme-metadata and
```
Comment (ndushay) [inline README.md]: are you suggesting that we have an 80 char line limit?  I'm not sure I agree.

File: README.md [hunk 2]
```diff
@@ -113,13 +120,14 @@ Configuration for transforms is specified in `config/metadata_mappings.json`. Fo
...
+This specifies that `mods_config.rb` is to be used for any files ending in
+`.mods` found in `stanford/maps`. `settings` are provided to the Traject indexer
+as additional settings.
 
 `extension` is optional; the default is `.xml`.
 
-Additional metadata mappings can be added to this file. In case a metadata file matches more than one configuration, the
-first one wins.
+Additional metadata mappings can be added to this file. In case a metadata file
+matches more than one configuration, the first one wins.
```
Comment (ndushay) [inline README.md]: Should we document when there are multiple values in "trajects"?  Like how stanford actually is in metadata_mappings?

File: README.md (hunk 1 above)
Comment (mjgiarlo) [reply README.md]: @ndushay Parts of the README respected an 80-char line limit and other parts did not. I thought it'd be good to be consistent. I'd be just as happy to remove the line limits across the board in the README.

**Comment by mjgiarlo** [comment]
> I wouldn't expect to find config files in a lib directory. What do you think about putting traject configs into a folder called ... `traject/configs` instead of `lib/configs` ? Or `traject_configs` ? Or `config/traject` ?

:100: fine with me. I prefer `traject_configs/`, I think?

File: README.md (hunk 2 above)
Comment (mjgiarlo) [reply README.md]: @ndushay I'm not sure I follow this one.

File: README.md (hunk 1 above)
Comment (ndushay) [reply README.md]: At this point, it is less work to keep them, but in general i'm happier removing them.  It's not important, really.

File: README.md (hunk 2 above)
Comment (ndushay) [reply README.md]:
```json
  {
    "trajects": [
      "mods_config.rb",
      "stanford_mods_config.rb"
    ],
    "paths": [
      "stanford/maps"
    ],
    "extension": ".mods",
    "settings": {
```

Comment (mjgiarlo) [reply README.md]: @ndushay ah! yes, done.

**Comment by jacobthill** [comment]
@mjgiarlo it looks like it can't find the configs ```[ERROR] No such file or directory @ rb_sysopen - traject_configs/mods_config.rb``` That path doesn't exist. The configs are in ```dlme-transform/lib/configs```

**Comment by mjgiarlo** [comment]
@jacobthill :speech_balloon: 
> @mjgiarlo it looks like it can't find the configs `[ERROR] No such file or directory @ rb_sysopen - traject_configs/mods_config.rb` That path doesn't exist. The configs are in `dlme-transform/lib/configs`

Oh, that's hilarious. I changed all the settings and then failed to move the configs. Ha! Change incoming.

**Comment by jacobthill** [comment]
Now getting the following error: ```[ERROR] Error loading configuration file traject_configs/mods_config.rb:99 NoMethodError:undefined method `include?' for nil:NilClass
/usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:231:in `rescue in block in load_config_file': Error loading configuration file traject_configs/mods_config.rb:99 NoMethodError:undefined method `include?' for nil:NilClass (Traject::Indexer::ConfigLoadError)
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:228:in `block in load_config_file'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:227:in `open'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:227:in `load_config_file'
	from /opt/traject/lib/transformer.rb:30:in `block (2 levels) in transformer'
	from /opt/traject/lib/transformer.rb:30:in `each'
	from /opt/traject/lib/transformer.rb:30:in `block in transformer'
	from /opt/traject/lib/transformer.rb:29:in `tap'
	from /opt/traject/lib/transformer.rb:29:in `transformer'
	from /opt/traject/lib/transformer.rb:22:in `transform'
	from /opt/traject/lib/cli.rb:78:in `block in transform_all'
	from /opt/traject/lib/cli.rb:72:in `each'
	from /opt/traject/lib/cli.rb:72:in `transform_all'
	from /opt/traject/lib/cli.rb:56:in `transform'
	from /usr/local/bundle/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /usr/local/bundle/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /usr/local/bundle/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /usr/local/bundle/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from exe/transform:8:in `<main>'```

**Comment by mjgiarlo** [comment]
@jacobthill :speech_balloon: 
> Some of the configs that have the cho_date_range_norm line commented out should pass validation but its not a problem. I can uncomment those when I'm ready to re-map the data.

OK, sounds good. I tried to comment them out one-by-one whenever the integration test failed, but perhaps my mapping of the output to the config was mistaken.

File: lib/config_finder.rb [hunk 3]
```diff
@@ -26,29 +29,44 @@ def find
 
     private
 
+    # TODO: Remove these exclusions when CI build is passing
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/MethodLength
     # Performs the mapping, returning a hash of data filepaths to config filenames
     def build_paths
       mapping = {}
+      puts "data_dir: #{data_dir.inspect}"
```
Comment (ndushay) [inline lib/config_finder.rb]: do we want to keep these puts statements?

Comment (mjgiarlo) [reply lib/config_finder.rb]: @ndushay no, those are in a fixup commit I'm going to remove once the CI build passes.

**Pull request merged**